lib/provider.js

```diff
@@ -101,12 +104,50 @@ Provider.prototype.send = function(payload, callback) {
     self.options.logger.log("   > " + JSON.stringify(payload, null, 2).split("\n").join("\n   > "));
   }
 
-  this.engine.sendAsync(payload, intermediary);
+  if (self.options.asyncRequestProcessing) {
+    self.engine.sendAsync(payload, intermediary);
+  } else {
+    self._queueRequest(payload, intermediary);
+  }
 };
 
 Provider.prototype.close = function(callback) {
   // This is a little gross reaching, but...
   this.manager.state.blockchain.close(callback);
 };
 
+Provider.prototype._queueRequest = function(payload, intermediary) {
+  if (!this._requestQueue) {
+    this._requestQueue = []
+  }
+
+  this._requestQueue.push({
+    payload: payload,
+    callback: intermediary
+  });
+
+  setImmediate(this._processRequestQueue)
+}
+
+Provider.prototype._processRequestQueue = function() {
+  const self = this;
+
+  if (self._requestInProgress) return
+
+  self._requestInProgress = true
+
+  let args = self._requestQueue.shift()
+
+  if (args) {
+    self.engine.sendAsync(args.payload, (err, result) => {
+      if (self._requestQueue.length > 0) {
+        setImmediate(self._processRequestQueue)
+      }
+      args.callback(err, result)
+      self._requestInProgress = false;
+    })
+  }
```

Comment:
Looks like you might need to negate `_requestInProgress` if `args` is undefined.